runway/cfngin/hooks/aws_lambda.py

```diff
@@ -464,6 +464,29 @@ def dockerized_pip(work_dir, client=None, runtime=None, docker_file=None,
         LOGGER.info('lambda.docker: %s', log.decode().strip())
 
 
+def _pip_has_no_color_option(python_path):
+    """Return boolean on whether pip is new enough to have --no-color option.
+
+    pip v10 introduced this option.
+
+    """
+    try:
+        pip_version_string = subprocess.check_output(
+            [python_path,
+             '-c',
+             'from __future__ import print_function;'
+             'import pip;'
+             'print(pip.__version__)']
+        )
+        if sys.version_info[0] > 2 and isinstance(pip_version_string, bytes):
+            pip_version_string = pip_version_string.decode()
+        if int(pip_version_string.split('.')[0]) > 10:
+            return True
+    except (AttributeError, ValueError, subprocess.CalledProcessError):
+        LOGGER.debug('Error checking pip version; assuming it to be pre-v10')
+    return False
```

Comment:
There are better and more accurate options available to get a python packages version list `pkg_resources` or `importlib.metadata`.

Using `importlib`, you can add an import block like this

```python
if sys.version_info < (3, 8):
    # importlib.metadata is standard lib for python>=3.8, use backport
    from importlib_metadata import version, PackageNotFoundError  # pylint: disable=E
else:
    from importlib.metadata import version, PackageNotFoundError  # pylint: disable=E
```

and reduce the logic needed to determine support to this

```suggestion
    try:
        return int(version('pip').split('.')[0]) >= 10
    except PackageNotFoundError:
        LOGGER.debug('Error checking pip version; assuming it to be pre-v10')
        return False
```

Reply:
The supplied `python_path` here comes from the `_zip_package` function, where it may be different than what's currently running this code. I don't see a safer path than using subprocess in the same way that _zip_package is, but I'm open to suggestions.

Reply:
right... I  guess we did provide the option to specify a custom python path.

>  I think we're likely to toss it entirely in the relatively near future

if we are going to toss this in the near future, i am fine leaving this as is